internal/api/v1/services/instance.go

```diff
@@ -255,44 +258,81 @@ func (s *InstanceService) handleInfrastructureDeletion(
 		return fmt.Errorf("no instances found for job %d", job.ID)
 	}
 
-	// Calculate how many instances to delete
-	numberOfInstancesToDelete := 0
+	// TODO: this section of code could be refactored to be unit tested
+	// Check if we have specific instance names to delete
+	var instancesToDelete []models.Instance
+	var specificNamesToDelete []string
+
+	// Collect specific instance names to delete
 	for _, instanceReq := range infraReq.Instances {
-		numberOfInstancesToDelete += instanceReq.NumberOfInstances
+		if instanceReq.Name != "" {
+			specificNamesToDelete = append(specificNamesToDelete, instanceReq.Name)
+		}
 	}
 
-	if numberOfInstancesToDelete <= 0 {
-		return fmt.Errorf("invalid number of instances to delete: %d", numberOfInstancesToDelete)
-	}
+	// If specific names are provided, find those instances
+	if len(specificNamesToDelete) > 0 {
+		fmt.Printf("ℹ️ Deleting specific instances by name: %v\n", specificNamesToDelete)
+
+		// Find instances with the specified names
+		for _, name := range specificNamesToDelete {
+			found := false
+			for _, instance := range instances {
+				if instance.Name == name {
+					instancesToDelete = append(instancesToDelete, instance)
+					found = true
+					break
+				}
+			}
+			if !found {
+				fmt.Printf("⚠️ Warning: Instance with name '%s' not found\n", name)
+			}
+		}
+
+		if len(instancesToDelete) == 0 {
+			return fmt.Errorf("none of the specified instances were found")
+		}
+	} else {
+		// No specific names provided, use FIFO order (oldest first)
+		// Calculate how many instances to delete
+		numberOfInstancesToDelete := 0
+		for _, instanceReq := range infraReq.Instances {
+			numberOfInstancesToDelete += instanceReq.NumberOfInstances
+		}
+
+		if numberOfInstancesToDelete <= 0 {
+			return fmt.Errorf("invalid number of instances to delete: %d", numberOfInstancesToDelete)
+		}
+
+		// Limit the number of instances to delete to the available ones
+		if numberOfInstancesToDelete > len(instances) {
+			fmt.Printf("⚠️ Warning: Requested to delete %d instances but only %d are available\n",
+				numberOfInstancesToDelete, len(instances))
+			numberOfInstancesToDelete = len(instances)
+		}
+
+		// Select only the oldest instances to delete
+		instancesToDelete = instances[:numberOfInstancesToDelete]
```

Comment:
This form of deletion might be dangerous, I suggest to delete only the instances whose names are specifically submitted by the user.

Reply:
this was the default implementation. Delete a number of instances provided by the user. 
I think that is still a useful feature. 
For example, a user might was to delete half the nodes to see the impact.

Reply:
it is kinda random based on the creation time. It would be more accurate if user explicitly submits the instance names to be deleted. I would remove this. as it might do an unwanted action.

Reply:
capturing here https://github.com/celestiaorg/talis/issues/97